src/emulator/apphosting/serve.ts

```diff
@@ -55,13 +90,17 @@ async function serve(
   for (const env of apphostingLocalConfig.environmentVariables) {
     environmentVariablesAsRecord[env.variable] = env.value!;
   }
+  await Promise.all(
+    apphostingLocalConfig.secrets?.map(async (secret) => {
```

Comment:
We should update the logic for merge:

https://github.com/firebase/firebase-tools/blob/bbc33ec1d3130893030daa8e0f17be11e01f9aef/src/apphosting/yaml.ts#L85

Where we remove secrets if they are overridden by an env var in apphosting.local.yaml

Otherwise, there is no way to override the secret to prevent it from fetching it during emulator start.